### DISCUSSION TITLE
Use of Exiv2 groups: Image[2|3], MpfInfo, SubImage(1-9) and SubThumb1

### DISCUSSION THREAD

**Discussion by postscript-dev** [postscript-dev/exiv2#4] [Q&A]
@kmilos, 
I posted a question on the chat-server ([here](https://matrix.to/#/!hVIpXqhWrWWcqnvuXp:matrix.org/$p65GGyn-LJanPoTxQNIMU5SuB9JNwDFmQTc8xfS6EYs?via=matrix.org&via=gnome.org&via=jeanclaudegraf.ch)) to find out about the Image[2|3], MpfInfo, SubImage(1-9) and SubThumb1 groups. I hope to add a sentence to the manpage explaining what they were used for. @clanmills suggested that you were more experienced with this and may be able to help.
Thanks in advance.

**Comment by kmilos** [comment]
AFAIK, ImageN and SubImageN correspond to IFDs and their [subIFDs](https://www.awaresystems.be/imaging/tiff/tifftags/subifds.html)  as defined in the TIFF specification, which many image formats use as their basis. I'm not sure if they appear in and what they map to in other containers.

Also not sure about SubThumb and MpfInfo without diving into the code...

**Comment by clanmills** [comment]
Curiously, a Krita user has opened an issue about ImageN.  https://github.com/Exiv2/exiv2/issues/1879

I suspect what the user is discussing is "multi-layer" tiff which is a multi-page tiff in which every page is rendered into the same frame buffer.  Multi-page Tiff is on the Wish List.  We don't have to engineering resources to take it on.  https://github.com/Exiv2/exiv2/issues/1466#issuecomment-811787134

I'm happy to be retired.  I suspect getting to bottom on what's being discussed by that user could quite an undertaking.

**Comment by kmilos** [comment]
Yes, it does look like a limited number of IFDs/subIFDs is currently supported.

**Comment by Wormnest** [comment]
Until yesterday, we used a SubIFD for TIFF images to store the thumbnail in GIMP. Due to longstanding Windows Explorer thumbnail handling (see [https://gitlab.gnome.org/GNOME/gimp/-/issues/3740](https://gitlab.gnome.org/GNOME/gimp/-/issues/3740) we changed this to letting exiv2 handle this (i.e. storing it in the second IFD of a TIFF).

From what I remember reading, SubIFD's can also be used to store so called pyramid images, where the IFD they belong to has the full scale image and the SubIFD's hold smaller versions at different sizes.

**Reply by kmilos** [reply]
FWIW, most TIFF based raw containers actually put the thumbnail as the root IFD image, and the full resolution one in the subIFD (and more previews at different resolutions). Seems to make most image and file browsers happy.

**Comment by postscript-dev** [comment]
Thanks for helping everyone, this is much clearer.
